provider/cmd/pulumi-gen-eks/main.go

```diff
@@ -1502,7 +1576,10 @@ func nodeGroupProperties(cluster, v2 bool) map[string]schema.PropertySpec {
 				"value, the latest available version is used.",
 		},
 		"instanceProfile": {
-			TypeSpec:    schema.TypeSpec{Ref: awsRef("#/resources/aws:iam%2FinstanceProfile:InstanceProfile")},
+			TypeSpec: schema.TypeSpec{
+				Ref:   awsRef("#/resources/aws:iam%2FinstanceProfile:InstanceProfile"),
+				Plain: true,
+			},
 			Description: "The ingress rule that gives node group access.",
 		},
```

Comment:
This is plain in the implementation:

https://github.com/pulumi/pulumi-eks/blob/c6f1fce98fa2683abe7ffbad7f166da99928fb83/nodejs/eks/nodegroup.ts#L261-L265